src/main/java/com/es/phoneshop/web/ProductDetailsPageServlet.java

```diff
@@ -104,14 +102,19 @@ private void redirectToProductDetailsPageWithSuccessMessage(HttpServletRequest r
         response.sendRedirect(redirectPath);
     }
 
-    private Number parseQuantityRequestParameter(HttpServletRequest request) throws ParseException {
+    private int parseQuantityRequestParameter(HttpServletRequest request)
```

Comment:
you can create separate service to parse quantity, it will allow to avoid code duplication